src/psycopmlutils/timeseriesflattener/flattened_dataset.py

```diff
@@ -289,30 +302,225 @@ def _validate_processed_arg_dicts(self, arg_dicts: list):
                 f"Didn't generate any features because: {warnings}",
             )
 
-    def _flatten_temporal_values_to_df_wrapper(self, kwargs_dict: Dict) -> DataFrame:
-        """Wrap flatten_temporal_values_to_df with kwargs for multithreading
-        pool.
+    def _get_feature(self, kwargs_dict: Dict) -> DataFrame:
+        """Get features. Either load from cache, or generate if necessary.
 
         Args:
             kwargs_dict (Dict): Dictionary of kwargs
 
         Returns:
             DataFrame: DataFrame generates with create_flattened_df
         """
-        return self.flatten_temporal_values_to_df(
-            prediction_times_with_uuid_df=self.df[
-                [
-                    self.pred_time_uuid_col_name,
-                    self.id_col_name,
-                    self.timestamp_col_name,
-                ]
-            ],
-            id_col_name=self.id_col_name,
-            timestamp_col_name=self.timestamp_col_name,
-            pred_time_uuid_col_name=self.pred_time_uuid_col_name,
-            **kwargs_dict,
+        full_col_str = generate_feature_colname(
+            prefix=kwargs_dict["new_col_name_prefix"],
+            out_col_name=kwargs_dict["new_col_name"],
+            interval_days=kwargs_dict["interval_days"],
+            resolve_multiple=kwargs_dict["resolve_multiple"],
+            fallback=kwargs_dict["fallback"],
+        )
+
+        file_pattern = f"{full_col_str}_{self.n_uuids}_uuids"
+
+        cache_hit = None
+
+        if hasattr(self, "feature_cache_dir"):
+            cache_hit = False
+
+            if self._cache_is_hit(
+                file_pattern=file_pattern,
+                full_col_str=full_col_str,
+                kwargs_dict=kwargs_dict,
+            ):
+                cache_hit = True
+
+                df = self._load_cached_df_and_expand_fallback(
+                    dir=self.feature_cache_dir,
+                    file_pattern=file_pattern,
+                    full_col_str=full_col_str,
+                    fallback=kwargs_dict["fallback"],
+                )
+
+                return df
+        else:
+            msg.info("No cache dir specified, not attempting load")
+
+        if not cache_hit:
```

Comment:
```suggestion
```
No need for the if - this should happen if the above doesn't run (the block above has a `return` statement so this won't run if the block above does